testing/releases/releases.go

```diff
@@ -68,10 +68,41 @@ func (d *Releases) ResolveURL(ctx context.Context, rq Requirements) (string, err
 	if res.StatusCode >= 300 {
 		return "", ErrNotFound
 	}
-	return d.decodeDownload(res.Body, rq)
+	r, err := d.decodeDownload(res.Body, rq)
+	if err != nil {
+		return "", err
+	}
+	return r[0], nil
+}
+
+// ResolveURLs gets the raw download URLs for all binaries of a release, based on the requirements (version, OS, arch)
+func (d *Releases) ResolveURLs(ctx context.Context, rq Requirements) (map[string]string, error) {
+	req, err := http.NewRequestWithContext(ctx, "GET", d.baseURL+"/"+rq.Version+"/checksums.txt", nil)
+	if err != nil {
+		return nil, err
+	}
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+	if res.StatusCode >= 300 {
```

Comment:
Is this the right status code range for this error? (Should it include 5xx?)

Reply:
maybe it should use ex http client